sklearn/calibration.py

```diff
@@ -885,29 +886,36 @@ def _sigmoid_calibration(
     T = np.zeros_like(y, dtype=np.float64)
     T[y > 0] = (prior1 + 1.0) / (prior1 + 2.0)
     T[y <= 0] = 1.0 / (prior0 + 2.0)
-    T1 = 1.0 - T
 
-    def objective(AB):
-        # From Platt (beginning of Section 2.2)
-        P = expit(-(AB[0] * F + AB[1]))
-        loss = -(xlogy(T, P) + xlogy(T1, 1.0 - P))
-        if sample_weight is not None:
-            return (sample_weight * loss).sum()
-        else:
-            return loss.sum()
+    bin_loss = HalfBinomialLoss()
 
-    def grad(AB):
-        # gradient of the objective function
-        P = expit(-(AB[0] * F + AB[1]))
-        TEP_minus_T1P = T - P
-        if sample_weight is not None:
-            TEP_minus_T1P *= sample_weight
-        dA = np.dot(TEP_minus_T1P, F)
-        dB = np.sum(TEP_minus_T1P)
-        return np.array([dA, dB])
+    def loss_grad(AB):
+        P = bin_loss.link.inverse(-(AB[0] * F + AB[1]))
+        raw_prediction = bin_loss.link.link(P)
```

Comment:
I think you had it right before: Just pass `raw_predictions=-(AB[0] * F + AB[1]))`
The minus is due to convention in Pratt‘s paper.

Reply:
Thanks for the review @lorentzenchr.

Reply:
I'm a bit worried why the tests still passed with wrong formulas.

Reply:
I think when we take the expit and then the logit we get the same value back. For example
```python
from scipy.special import expit, logit
A = -1.5
P = expit(A)
Q = logit(P)
>>Q
-1.5

A = 0.51
P = expit(A)
Q = logit(P)
>>Q
0.51
```